protocols/kad/src/behaviour.rs

```diff
@@ -761,6 +1008,24 @@ where
     }
 }
 
+/// The result of a `GET_VALUE` query.
+#[derive(Debug, Clone, PartialEq)]
+pub enum GetValueResult {
+    /// The results received from peers.
+    Found { results: Vec<Record> },
```

Comment:
Since you didn't like my suggestion for splitting the results into head/tail, shouldn't it at least be documented that `results` is never empty here? It is otherwise not clear to the user at all whether `results` can be empty and if so, what to do about it.